mesa/batchrunner.py

```diff
@@ -118,7 +118,7 @@ def collect_model_vars(self, model):
     def collect_agent_vars(self, model):
         """ Run reporters and collect agent-level variables. """
         agent_vars = {}
-        for agent in model.schedule.agents:
+        for agent in model.schedule.agents.values():
```

Comment:
I agree that preserving order is important. I'm not sure whether in Python <3.6 iterating over `values()` is guaranteed to preserve order any better than just iterating over the keys, and/or for that order to be the same as the order in which the agents were added (which should be the default non-random behavior).